### DISCUSSION TITLE
Electron apps crash when trying to upload files

### DISCUSSION THREAD

**Discussion by haarp** [netblue30/firejail#4405] [Q&A]
Hello,

so I've been having this problem for a while now. I run a few Electron apps (MS Teams, Discord, Signal) with firejail. All of them work fine for the most part, but will crash shortly after attempting to upload files.

- Start app
- Put file somewhere app can access (`$HOME/Downloads`)
- Open upload dialog in app and select file
- App crashes shortly after selecting the file, or as soon as you press another key to write a comment. If you're quick enough, sometimes you can upload and post the file before the crash.

There doesn't seem to be anyone else having this problem, or at least my Google-Fu failed me. The behavior is very similar across all three of these apps. I'm not even sure it's firejail-related. But I would not dare run these closed blobs without sandboxing.

Here's some log from Discord. Maybe something related to the Gtk file picker?
```
(discord:39): Gtk-WARNING **: 00:15:43.208: Symlink resolution failed: Failed to read the symbolic link “/home/haarp/.local/share/recently-used.xbel”: No such file or directory

(discord:39): Gtk-WARNING **: 00:15:43.208: Attempting to set the permissions of '(invalid filename)', but failed: Bad address

(discord:39): Gtk-WARNING **: 00:15:43.208: Attempting to store changes into '(invalid filename)', but failed: (null)

***MEMORY-ERROR***: discord[39]: GSlice: assertion failed: sinfo->n_allocated > 0
Cannot upload crash dump: cannot exec /usr/bin/wget

Unexpected crash report id length
Failed to get crash dump id.
Report Id:

Parent is shutting down, bye...
```

firejail-0.9.64.4, using mostly the default profiles (at least nothing that should cause such problems).

Any ideas? Thanks!

edit:
Ok, it's definitely related to the GTK portion of Electron. Running with `G_SLICE=debug-blocks`:

`GSlice: MemChecker: attempt to release non-allocated block: 0xfoo size=16`

**Comment by rusty-snake** [comment]
I've no idea so far, but is anything relevant in the syslog?

**Reply by haarp** [reply]
indeed.

```
kernel: Discord[7964]: segfault at ... ip ... sp ... error 4 in libglib-2.0.so.0.6800.3[...]
kernel: Code: 44 8d 78 ff 48 8b 0b 4d 89 fe 49 c1 e6 04 4c 01 f1 4c 8b 21 48 89 c5 4d 85 e4 0f 84 ec 00 00 00 49 8b 44 24 08 48 85 c0 74 52 <48> 8b 10 49 89 54 24 08 49 89 c4 48 8b 41 08 48 85 c0 75 2e 66 0f
```

which further points towards Gtk/glib being the problem. The "code" is the only thing that stays constant between crashes. Doesn't seem to be firejail, does it?

edit: It's firejail. Discord doesn't crash when run directly.
